drivers/flash/nrf_qspi_nor.c

```diff
@@ -332,6 +346,63 @@ static void qspi_handler(nrfx_qspi_evt_t event, void *p_context)
 	}
 }
 
+static int anomaly_122_init(const struct device *dev)
+{
+	int ret = 0;
+
+#ifdef CONFIG_NORDIC_QSPI_NOR_NRF52840_ANOMALY_122
+	struct qspi_nor_data *dev_data = get_dev_data(dev);
+	bool first = true;
+	nrfx_err_t res;
+
+	k_msleep(50);
+	qspi_lock(dev);
+
+#ifdef CONFIG_MULTITHREADING
```

Comment:
would be great to have comment which clarifies why workaround uses dedicated locking mechanism.
I know the idea behind that from the commit message, but at the firs glance it looks like it would be possible use currently available locks. Pleas put comment here - will will avoid situation when someone will mess implemented behavior.

Reply:
Sure, done. Is it good?